doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,211 @@
+the fitting results might at times return values which do not make sense
+physically. (Negative f values or D*)
+"""
+
+ivimmodel = IvimModel(gtab)
+
+"""
+To fit the model, call the `fit_method` and pass the data for fitting.
+"""
+
+ivimfit = ivimmodel.fit(data_slice)
+
+"""
+The fit method creates a IvimFit object which contains the
+fitting parameters of the model the fit parameters of the model.
+These are accessible through the model_parameters attribute of the
+IvimFit object. Parameters are arranged as a 4D array, corresponding
+to the spatial dimensions of the data, and the last dimension (of length 4)
+corresponding to the model parameters according to the following
+order: S0, f, D* and D.
+"""
+
+ivimparams = ivimfit.model_params
+print("ivimparams.shape : {}".format(ivimparams.shape))
+
+"""
+As we see, we have a 20x20 slice at the height z = 33. Thus we
+have 400 voxels. We will now plot the values of S0, f, D* and D
+for some voxels and also the various maps for the entire slice.
+This will give us an idea about the diffusion and perfusion in
+the section. Let(i, j) denote the coordinate of the voxel.
+"""
+
+i, j = 10, 10
+estimated_params = ivimfit.model_params[i, j, :]
+print(estimated_params)
+
+"""
+Let us define a plotting our results. For this we will use the
+ivim_function defined in the module ivim which takes bvalues
+and ivim parameters and returns the estimated signal.
+"""
+
+estimated_signal = ivim_function(estimated_params, gtab.bvals)
+
+plt.scatter(gtab.bvals, data_slice[i, j, :], color="green",
+            label="Actual signal")
+plt.scatter(gtab.bvals, estimated_signal,
+            color="red", label="Estimated Signal")
+plt.xlabel("bvalues")
+plt.ylabel("Signals")
+
+S0_est, f_est, D_star_est, D_est = estimated_params
+text_fit = """Estimated \n S0={:06.3f} f={:06.4f}\n
+            D*={:06.5f} D={:06.5f}""".format(S0_est, f_est, D_star_est, D_est)
+
+plt.text(0.65, 0.50, text_fit, horizontalalignment='center',
+         verticalalignment='center', transform=plt.gca().transAxes)
+plt.legend(loc='upper left')
+plt.savefig("ivim_voxel_plot.png")
+plt.close()
+
+"""
+Now we can map the perfusion and diffusion maps for the slice. We
+will plot a heatmap showing the values using a colormap. It will be
+useful to define a plotting function for the heatmap here since we
+will use it to plot for all the IVIM parameters. We will need to specify
+the lower and upper limits for our data. For example, the perfusion
+fractions should be in the range (0,1). Similarly, the diffusion and
+pseudo-diffusion constants are much smaller than 1. We pass an argument
+called "variable" to out plotting function which gives the label for
+the plot.
+"""
+
+
+def plot_map(raw_data, variable, limits):
+    lower, upper = limits
+    plt.title('Map for {}'.format(variable))
+    plt.imshow(raw_data.T, origin='lower', clim=(lower, upper), cmap="gray")
+    plt.colorbar()
+    plt.savefig(variable + ".png")
```

Comment:
These file names are not appropriate. Some of them have symbols ("*"), and some of them have spaces in them. Please change this, so that we get reasonable file-names.